packages/slate-react/src/plugins/debug.js

```diff
@@ -0,0 +1,64 @@
+import Debug from 'debug'
+
+/**
+ * A plugin that adds the "before" browser-specific logic to the editor.
+ *
+ * @return {Object}
+ */
+
+function DebugPlugin(namespace) {
+  /**
+   * Debug.
+   *
+   * @type {Function}
+   */
+
+  const debug = Debug(namespace)
+
+  const events = [
+    'onBeforeInput',
+    'onBlur',
+    'onClick',
+    'onCompositionEnd',
+    'onCompositionStart',
+    'onCopy',
+    'onCut',
+    'onDragEnd',
+    'onDragEnter',
+    'onDragExit',
+    'onDragLeave',
+    'onDragOver',
+    'onDragStart',
+    'onDrop',
+    'onFocus',
+    'onInput',
+    'onKeyDown',
+    'onPaste',
+    'onSelect',
+  ]
+
+  const plugin = {}
+
+  for (const eventName of events) {
+    plugin[eventName] = function(event, editor, next) {
+      debug(eventName, { event })
```

Comment:
This plugin should be removed from the default plugins in `plugins/dom.js`.

One question I have is whether we might want to leave the library in the code as it is valuable during debugging. I relied on this when developing the Android plugin. The reason is it allowed me to see when an event passed through a plugin. For example, I can see if the android plugin allowed or disallowed an event from passing through to the `before` plugin. The debug statements in the `before` plugin don't tell us this definitively as sometimes they return before the `debug` message gets called.

Also, sometimes I would place this debug plugin after the `before` plugin to see what made it through there as well.

If I'm working on Android again, I'd probably add it back in for the duration of my work and then remove it when I'm done. It would be simpler if it wasn't in a separate NPM package which I have to add to `package.json` and remove.

I've removed it from `plugins/dom` just now (Done). Please let me know what you'd prefer about keeping or removing the library as a whole. Thanks.

Reply:
@thesunny that makes sense, I'm okay with keeping it in the source here. And potentially we can improve it over time to make it more useful UX wise.

Reply:
Acknowledged. Left it in.